### PR TITLE
Replace (black|white)list with more inclusive language

### LINKED ISSUE #7147: Refrain from using terms such as blacklist and whitelist in the codebase
**Describe the bug**
In some parts of the codebase some fields can still be found that use the term `blacklist` or `whitelist`, DEV is a very open and diverse community and I think the codebase should reflect that.

**To Reproduce**

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**

- OS:
- Browser:
- Version:

**Smartphone (please complete the following information):**

- Device:
- OS:
- Browser:
- Version:

**Additional context**
I wrote this using the bug template as I don't think this should be a feature and also because I do not think that this meant discriminatory by any means by any of the DEV staff or community.

### PR DESCRIPTION
**Pull request by rhymes**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [x] Documentation Update

## Description

This is an attempt to curb the usage of words like `whitelist`/`blacklist` in our code.

The only one instances remaining are from third party gems.

## Related Tickets & Documents

Closes #7147

## Added tests?

- [ ] yes
- [x] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [x] docs.dev.to
- [ ] readme
- [ ] no documentation needed

## [optional] Are there any post deployment tasks we need to perform?

We need to set `FASTLY_SAFE_PARAMS_SNIPPET_NAME` to the value that was previously set to `FASTLY_WHITELIST_PARAMS_SNIPPET_NAME`

### CODE REVIEW

File: app/assets/javascripts/base.js.erb [hunk 1]
```diff
@@ -57,7 +57,7 @@ var instantClick
     return target
   }
 
-  function isBlacklisted(elem) {
+  function isForbiddenFromPreloading(elem) {
```
Comment (Ridhwana) [inline app/assets/javascripts/base.js.erb:60]: @rhymes this is purely a personal opinion but this function name seems a bit clunky, what about something like `isForbiddenFromPreload`  or `isNotPreloadable`

Comment (rhymes) [reply app/assets/javascripts/base.js.erb:60]: Make sense, i'll change it! thank you @Ridhwana!

File: spec/tasks/fastly_spec.rb [hunk 2]
```diff
@@ -6,14 +6,16 @@
...
     it "doesn't run if Fastly isn't configured" do
-      allow(ApplicationConfig).to receive(:[]).with("FASTLY_API_KEY").and_return(nil)
-      allow(ApplicationConfig).to receive(:[]).with("FASTLY_SERVICE_ID").and_return(nil)
-      allow(ApplicationConfig).to receive(:[]).with("FASTLY_WHITELIST_PARAMS_SNIPPET_NAME").and_return(nil)
-      allow(FastlyVCL::WhitelistedParams).to receive(:update)
-      Rake::Task["fastly:update_whitelisted_params"].invoke
-      expect(FastlyVCL::WhitelistedParams).not_to have_received(:update)
+      %w[FASTLY_API_KEY FASTLY_SERVICE_ID FASTLY_SAFE_PARAMS_SNIPPET_NAME].each do |var|
+        allow(ApplicationConfig).to receive(:[]).with(var).and_return(nil)
+      end
```
Comment (atsmith813) [inline spec/tasks/fastly_spec.rb:13]: +1

File: spec/tasks/fastly_spec.rb [hunk 3]
```diff
@@ -6,14 +6,16 @@
...
+      allow(FastlyVCL::SafeParams).to receive(:update)
+
+      Rake::Task["fastly:update_safe_params"].invoke
```
Comment (atsmith813) [inline spec/tasks/fastly_spec.rb:16]: We need to fix this or comment it out

https://github.com/thepracticaldev/dev.to/pull/7486

Comment (rhymes) [reply spec/tasks/fastly_spec.rb:19]: Done! See https://github.com/thepracticaldev/dev.to/pull/7459/commits/da890c7d91c3ce62b8f4f2727bb9a3754a76c652

**Comment by atsmith813** [comment]
@rhymes the conflict is now from the _actual_ fix for Travis in https://github.com/thepracticaldev/dev.to/pull/7487. This should be all wrapped up now 💯

**Comment by rhymes** [comment]
@atsmith813 waiting to merge this as we need to set the env var and I can't do it

**Comment by vaidehijoshi** [comment]
Really glad to see that we're doing this 👍

**Pull request merged**